### PR TITLE
Use Code.compiler_options[:no_warn_undefined]

### LINKED ISSUE #9192: Do not emit undefined warnings for entries in Code.compiler_options[:no_warn_undefined]

### PR DESCRIPTION
**Pull request by ericmj**
Closes #9192.

### CODE REVIEW

File: lib/elixir/lib/module/checker.ex [hunk 1]
```diff
@@ -47,6 +47,14 @@ defmodule Module.Checker do
     |> emit_warnings()
   end
 
+  defp no_warn_undefined(module_map) do
+    for(
+      {:no_warn_undefined, values} <- module_map.compile_opts,
+      value <- List.wrap(values),
+      do: value
+    ) ++ Code.compiler_options()[:no_warn_undefined]
+  end
```
Comment (josevalim) [inline lib/elixir/lib/module/checker.ex]: By doing this we are doing some premature work for something we may never use. My suggestion is to compute this on `no_warn_undefined?`.

File: lib/elixir/lib/module/checker.ex [hunk 2]
```diff
@@ -34,7 +34,7 @@ defmodule Module.Checker do
       cache: cache,
       file: module_map.file,
       module: module_map.module,
-      compile_opts: module_map.compile_opts,
+      no_warn_undefined: no_warn_undefined(module_map),
```
Comment (josevalim) [inline lib/elixir/lib/module/checker.ex]: Given the below, I would keep this as `compile_opts` still.

File: lib/elixir/lib/module/checker.ex (hunk 1 above)
Comment (ericmj) [reply lib/elixir/lib/module/checker.ex]: OTOH we will have to recompute it for every warning and skipped warning. Right now we are only precomputing it once per module.

Comment (josevalim) [reply lib/elixir/lib/module/checker.ex]: Good point. Sounds good to me then.

**Pull request merged**